documentation/design/user-stories/shared-infra.adoc

```diff
@@ -0,0 +1,277 @@
+== Enable messaging for a namespace by creating a tenant and an address
+
+Messaging tenant defines the following resources:
+
+```
+apiVersion: enmasse.io/v1beta2
+kind: MessagingTenant
+metadata:
+  name: default
+  namespace: myapp
+spec: {}
+---
+apiVersion: enmasse.io/v1beta2
+kind: MessagingAddress
```

Comment:
@lulf so what should happen when I create MessagingAddress CR in namespace where MessagingTenant is not deployed? Does it create messaging tenant default in namespace, or it should fail?

Reply:
@kornys I was thinking it would report in the address status that it cannot find a messaging tenant, and address will be in phase Pending (or Failed?).

Reply:
@lulf maybe keep it in Pending phase with error message, and I expect that when I create MessagingTenant in namespace, address will be configured and CR switch to Active phase? WDYT?

Reply:
or maybe would be nice if operator creates messaging tenant if does not exist

Reply:
> @lulf maybe keep it in Pending phase with error message, and I expect that when I create MessagingTenant in namespace, address will be configured and CR switch to Active phase? WDYT?

Yes, that is what I had envisioned.

> or maybe would be nice if operator creates messaging tenant if does not exist

We discussed this when going through these user stories initially, and found that automatically creating the messaging tenant would make it a bit strange wrt. what happens when you remove all addresses. I.e. it would either have to remain or we would have to delete it when there are no address. The behavior gets a bit messy, so I think the cleanest way is to require it to be created, which shouldn't be much of a problem for the 'getting started' experience.

Reply:
Hope it won't be confusing for users :)

Reply:
Yeah, I'm open to change behavior here based on what we see, but I think explicit creation is a better initial default than auto-create to avoid accidental confusion.